src/components/Select/Select.tsx

```diff
@@ -266,7 +277,23 @@ export const Select: FC<SelectProps> = React.forwardRef(
       if (readonly && dropdownVisible) {
         setDropdownVisibility(false);
       }
-    }, [readonly, dropdownVisible]);
+      if (!dropdownVisible && prevDropdownVisible) {
+        inputRef.current?.focus();
+      }
+    }, [readonly, dropdownVisible, prevDropdownVisible]);
+
+    useEffect(() => {
+      if (!filterable && !initialFocus) {
+        setInitialFocus(true);
+      } else if (!filterable && initialFocus !== null) {
+        setInitialFocus(initialFocus);
+      }
+      if (filterable && !initialFocus) {
+        setInitialFocus(false);
+      } else if (filterable && initialFocus !== null) {
+        setInitialFocus(initialFocus);
+      }
+    }, [filterable, initialFocus]);
```

Comment:
this has an initialFocus dependency and is updating initialFocus value. Any chance this is problematic?

Reply:
nevermind... see there is internal _initialFocus being updated